### DISCUSSION TITLE
How to clear and replace multiple tables safely?

### DISCUSSION THREAD

**Discussion by MaleMaldives** [ujjwalguptaofficial/JsStore#283] [Q&A]
I have my data in multiple tables, and I need to clear each one and insert new data. The problem is if the page is interrupted in the middle of saving two tables, then some tables will have new information and the other old (or none at all if between clear and insert).

Is there a way to clear and insert in one command? Is there way to do that for multiple tables in one command? The only other solution I can think of is to have a back up database, and alternate saving between the two, and in a third(tiny) database, keep track of which is the newest. But that doubles the data.

**Comment by ujjwalguptaofficial** [comment]
JsStore call query in a linear manner, so if you call three apis - `clear`, `insert` , `select`, clear will be executed first and then insert and in the last select.

So you just need to call right set of api one after another. In your case - its `clear` and then `insert` .
